configure.ac

```diff
@@ -142,25 +142,26 @@ AS_IF([test "x$have_jsoncpp" = "xyes"], [
 ])
 
 AC_ARG_WITH([openssl], AS_HELP_STRING([--with-openssl], [Build with OpenSSL support]))
-AS_IF([test "x$with_openssl" != "xno"],
+AS_IF([test "x$with_openssl" = "xyes"],
       [PKG_CHECK_MODULES([OpenSSL], [openssl >= 1.1], [have_openssl=yes], [have_openssl=no])],
       [have_openssl=no])
 AS_IF([test "x$have_openssl" = "xyes"], [
+    LDFLAGS="${LDFLAGS} -lssl"
     AC_MSG_NOTICE([Using OpenSSL])
 ], [
     AC_MSG_NOTICE([Not using OpenSSL])
 ])
 
 AC_ARG_WITH([http_parser_fork], AS_HELP_STRING([--with-http-parser-fork], [Build with http_parser fork to support old API]))
-AS_IF([test "x$with_http_parser_fork" != "xno"],[
+AS_IF([test "x$with_http_parser_fork" = "xyes"],[
     AC_MSG_NOTICE([Using http_parser fork])
-    CPPFLAGS+=" -DOPENDHT_PROXY_HTTP_PARSER_FORK"
+    CPPFLAGS+=" -DOPENDHT_PROXY_HTTP_PARSER_FORK=1"
```

Comment:
should not be necessary

Reply:
Done at ad695035aa25ff725c51447a3aa5fb5241e9cfe2